grails-app/services/org/pih/warehouse/importer/InventoryImportDataService.groovy

```diff
@@ -339,23 +339,26 @@ class InventoryImportDataService implements ImportDataService {
     private InventoryItem parseInventoryItem(Product product, def lotNumberRaw, def expirationDateRaw) {
         String lotNumber = lotNumberRaw instanceof Double ? lotNumberRaw.toInteger() : lotNumberRaw
 
-        Date expirationDate = null
+        if (!expirationDateRaw) {
+            return inventoryService.findAndUpdateOrCreateInventoryItem(product, lotNumber, null)
+        }
+
         if (expirationDateRaw instanceof String) {
-            expirationDate = EXPIRATION_DATE_FORMAT.parse(expirationDateRaw)
+            Date expirationDate = EXPIRATION_DATE_FORMAT.parse(expirationDateRaw)
             Calendar calendar = Calendar.getInstance()
             calendar.setTime(expirationDate)
-            expirationDate = calendar.getTime()
-        } else if (expirationDateRaw instanceof Date) {
-            expirationDate = expirationDateRaw
-        } else if (expirationDateRaw instanceof LocalDate) {
-            expirationDate = expirationDateRaw.toDate()
+            return inventoryService.findAndUpdateOrCreateInventoryItem(product, lotNumber, calendar.getTime())
+        }
+
+        if (expirationDateRaw instanceof LocalDate) {
+            return inventoryService.findAndUpdateOrCreateInventoryItem(product, lotNumber, expirationDateRaw.toDate())
         }
 
-        return inventoryService.findAndUpdateOrCreateInventoryItem(product, lotNumber, expirationDate)
+        return inventoryService.findAndUpdateOrCreateInventoryItem(product, lotNumber, expirationDateRaw)
     }
 
     private Location parseBinLocation(String binLocationName, Location parentLocation) {
-        if (binLocationName == null) {
+        if (!binLocationName) {
```

Comment:
~~What about if we have a bin location with a "" name? I know we've had issues with blanks vs nulls before so I'd be weary of treating them both as equivalent to null.~~

Nvm I checked and locations cannot be blank so this is fine.